drivers/i2c/i2c_omap.c

```diff
@@ -315,8 +315,7 @@ static void i2c_omap_resize_fifo(const struct device *dev, uint8_t size)
  */
 static int i2c_omap_get_sda(void *io_context)
 {
-	const struct i2c_omap_cfg *cfg = (const struct i2c_omap_cfg *)io_context;
-	i2c_omap_regs_t *i2c_base_addr = (i2c_omap_regs_t *)cfg->base.addr;
+	i2c_omap_regs_t *i2c_base_addr = DEV_I2C_BASE(io_context);
```

Comment:
Does this compile? It looks like it de-references a void pointer (`io_context->data`)
```suggestion
        const struct device *dev = io_context;
	i2c_omap_regs_t *i2c_base_addr = DEV_I2C_BASE(dev);
```

Reply:
I did not get any errors while compiling (maybe some casting going on somewhere). But I have updated the PR since it does seem better to be deliberate.